migrator.py

```diff
@@ -349,17 +367,26 @@ def migrate_variables():
                 if workspace_name not in workspaces and "*" not in workspaces:
                     continue
 
-                if fetch_scalr(
+                workspace_exists = fetch_scalr(
                     "workspaces",
-                    {"filter[name]": workspace_name, "filter[environment]": env["id"]}
-                )["data"]:
+                    {"filter[name]": workspace_name, "filter[environment]": env["id"]},
+                )["data"]
+                # workspace must exist if skip_workspace_creation
+                # workspace must not exist if not skip_workspace_creation
+                if len(workspace_exists) ^ skip_workspace_creation:
```

Comment:
we prefer using `and` instead of `^`
```suggestion
                if len(workspace_exists) and skip_workspace_creation:
```

Reply:
Not sure what you mean, `or` is not a replacement for `^`. The statement with XOR is equivalent to `if (not len(workspace_exists) and skip_workspace_creation) or (len(workspace_exists and not skip_workspace_creation)):`. I was thinking the XOR was a little simpler, let me know what you think.